relay-server/src/services/spooler/sql.rs

```diff
@@ -70,11 +70,11 @@ pub fn delete<'a>(key: QueueKey) -> Query<'a, Sqlite, SqliteArguments<'a>> {
         .bind(key.sampling_key.to_string())
 }
 
-/// Creates a query which fetches the `envelopes` table size.
+/// Creates a query which estimates the `envelopes` table size.
 ///
-/// This info used to calculate the current allocated database size.
-pub fn current_size<'a>() -> Query<'a, Sqlite, SqliteArguments<'a>> {
-    sqlx::query(r#"SELECT SUM(pgsize - unused) FROM dbstat WHERE name="envelopes""#)
+/// This info used to estimate the current allocated database size.
+pub fn estimate_size<'a>() -> Query<'a, Sqlite, SqliteArguments<'a>> {
+    sqlx::query(r#"SELECT SUM(length(envelope)) FROM envelopes"#)
```

Comment:
This way we still need to iterate the table ,
what if we actually revert to the initial query 
```rust
sqlx::query(
        "SELECT page_count * page_size as size FROM pragma_page_count(), pragma_page_size();",
    )
```

which should be much faster but also not exact number , but estimate

Reply:
Sounds good, the `sum(length())` is pretty fast, but the page count/size is still a bit faster, the values are very similar:

```
sum(length(envelope)): 492855289
page_size * page_count: 491520000
dbstat: 507478016
```

Reply:
This breaks for small spools (in tests):

```
dbstat:                               3308
sum(length(envelope)):   1600
page_size * page_count: 16384
```